src/com/jedk1/jedcore/ability/earthbending/MetalFragments.java

```diff
@@ -246,20 +242,21 @@ public void progress() {
 				fb.getLocation().getBlock().setBlockData(fb.getBlockData());
 			}
 		}
+		for (ListIterator<Item> iterator = thrownFragments.listIterator();iterator.hasNext();){
+			Item f = iterator.next();
 
-		for (Item f : thrownFragments) {
-			if (f.isOnGround())
-				f.remove();
-
-			if (f.isDead())
-				continue;
-
+			boolean touchedLiving = false;
 			for (Entity e : GeneralMethods.getEntitiesAroundPoint(f.getLocation(), 1)) {
 				if (e instanceof LivingEntity && e.getEntityId() != player.getEntityId()) {
+					touchedLiving = true;
 					DamageHandler.damageEntity(e, damage, this);
-					f.remove();
 				}
 			}
+			if(touchedLiving||f.isOnGround()||f.isDead()) {
```

Comment:
can you conform to the standard style a little more in the future? with spaces between the parts of this if statement

Reply:
Here I was scared, that I will ruin imports again with Alt+Ctrl+L :D 
Now I know, that I can use reformatting on only highlighted part of code.